api/services/device.go

```diff
@@ -149,10 +164,39 @@ func (s *service) UpdatePendingStatus(ctx context.Context, uid models.UID, statu
 		return NewErrDeviceStatusAccepted(nil)
 	}
 
+	ns, err := s.store.NamespaceGet(ctx, tenant)
+	if err != nil {
+		return NewErrNamespaceNotFound(tenant, err)
+	}
+
 	if status != StatusAccepted {
 		return s.store.DeviceUpdateStatus(ctx, uid, status)
 	}
 
+	// NOTICE: The logic below is only executed when the new status is "accepted".
+
+	if ns.MaxDevices > 0 {
+		slot, err := s.store.DeviceRemovedGet(ctx, tenant, uid)
+		if err != nil && err != store.ErrNoDocuments {
+			return NewErrDeviceRemovedGet(err)
+		}
+
+		if slot == nil {
```

Comment:
I'd handle the `if slot != nil` case first. However, adding a comment explaining it is good as well.